src/host/ut_host/SelectionTests.cpp

```diff
@@ -236,69 +251,90 @@ class SelectionTests
 
-        VerifyGetSelectionRects_LineMode();
+            VerifyGetSelectionRects_LineMode();
+        }
 
-        // #4 bottom-right to top-left selection
-        m_pSelection->_coordSelectionAnchor.x = m_pSelection->_srSelectionRect.right;
-        m_pSelection->_coordSelectionAnchor.y = m_pSelection->_srSelectionRect.bottom;
+        {
+            auto selection{ m_pSelection->_d.write() };
+            // #3 bottom-left to top-right selection
+            selection->coordSelectionAnchor.x = selection->srSelectionRect.left;
+            selection->coordSelectionAnchor.y = selection->srSelectionRect.bottom;
 
-        VerifyGetSelectionRects_LineMode();
+            VerifyGetSelectionRects_LineMode();
+        }
 
-        // Part II: Single line selection
-        m_pSelection->_srSelectionRect.top = 2;
-        m_pSelection->_srSelectionRect.bottom = 2;
-        m_pSelection->_srSelectionRect.left = 1;
-        m_pSelection->_srSelectionRect.right = 10;
+        {
+            auto selection{ m_pSelection->_d.write() };
+            // #4 bottom-right to top-left selection
+            selection->coordSelectionAnchor.x = selection->srSelectionRect.right;
+            selection->coordSelectionAnchor.y = selection->srSelectionRect.bottom;
 
-        // #1: left to right selection
-        m_pSelection->_coordSelectionAnchor.x = m_pSelection->_srSelectionRect.left;
-        VERIFY_IS_TRUE(m_pSelection->_srSelectionRect.bottom == m_pSelection->_srSelectionRect.top);
-        m_pSelection->_coordSelectionAnchor.y = m_pSelection->_srSelectionRect.bottom;
+            VerifyGetSelectionRects_LineMode();
+        }
 
-        VerifyGetSelectionRects_LineMode();
+        {
+            auto selection{ m_pSelection->_d.write() };
+            // Part II: Single line selection
+            selection->srSelectionRect.top = 2;
+            selection->srSelectionRect.bottom = 2;
+            selection->srSelectionRect.left = 1;
+            selection->srSelectionRect.right = 10;
+
+            // #1: left to right selection
+            selection->coordSelectionAnchor.x = selection->srSelectionRect.left;
+            VERIFY_IS_TRUE(selection->srSelectionRect.bottom == selection->srSelectionRect.top);
+            selection->coordSelectionAnchor.y = selection->srSelectionRect.bottom;
+
+            VerifyGetSelectionRects_LineMode();
+        }
 
-        // #2: right to left selection
-        m_pSelection->_coordSelectionAnchor.x = m_pSelection->_srSelectionRect.right;
-        VERIFY_IS_TRUE(m_pSelection->_srSelectionRect.bottom == m_pSelection->_srSelectionRect.top);
-        m_pSelection->_coordSelectionAnchor.y = m_pSelection->_srSelectionRect.top;
+        {
+            auto selection{ m_pSelection->_d.write() };
+            // #2: right to left selection
+            selection->coordSelectionAnchor.x = selection->srSelectionRect.right;
+            VERIFY_IS_TRUE(selection->srSelectionRect.bottom == selection->srSelectionRect.top);

```

Comment:
what even *is* this check? i mean, i guess it's defending the tests against a bad test author. but _we set `bottom` ourselves_.